lib/open_api_spex/open_api.ex

```diff
@@ -80,10 +80,11 @@ defmodule   OpenApiSpex.OpenApi do
           |> to_json()
         end
         defp to_json(value) when is_map(value) do
+          extensions = to_json(value[:extensions]) || %{}
```

Comment:
It feels like this might clobber a property named `extensions` defined in a schema?

I wonder if instead of providing a struct key for extensions, we could provide a function `add_extension` which would just `Map.put` the additional attribute into the struct?

Reply:
I do it like this now:

```elixir
 def spec do
    %OpenApi{
      info: %Info{
        title: "Company Platform API",
        version: "2.0.0",
        extensions: %{
          "x-logo" => %{
            "url" => "https://example.com/logo.png",
            "altText" => "Company logo"
          }
        },
        description: """
...
```

I see that having `add_extension` function still requires to extend `Info` structure.

Reply:
I prefer the `extensions` struct key over a function. A function is more complex, there is more room for things to go wrong.

I think the cause of this clobbering problem is that the encoder module potentially encodes any data, and not just the structs defined by the package. If this encoder only encoded structs from this package, then there wouldn't be a problem. Perhaps the encoder can be refactored to only encode the structs from open_api_spex?

Reply:
How about a protocol that we can opt-in for OpenApiSpex structs?

```elixir
  defprotocol Extendable do
    @fallback_to_any true
    def to_map(struct)
  end

  defimpl Extendable, for: Any do
    def to_map(struct), do: Map.from_struct(struct)
  end

  defimpl Extendable, for: [Info] do # We can add more struct types to this list over time as needed
    def to_map(struct = %{extensions: e}) do
      struct
      |> Map.from_struct()
      |> Map.delete(:extensions)
      |> Map.merge(e || %{})
    end
  end
```

Then when converting to JSON:

```elixir
        defp to_json(value = %{__struct__: _}) do
          value
          |> Extendable.to_map()
          |> to_json()
        end
```

Reply:
Hi @mbuhot, it looks good. I'll rework this PR soon.